### PR TITLE
Relicense to from LGPL-3 to BSD-3

### LINKED ISSUE #5148: Re-licensing iris from LGPL to BSD-3
To enable wider adoption of Iris, we are working on re-licensing it to be the more flexible BSD-3 license. 

Before we can do this we need all previous contributors to sign the latest SciTools Contributors License Agreement (CLA) as this includes a clause stating that the SciTools libraries can be re-licensed to any other OSI approved license. 

If your username appears below, could you please sign the updated CLA (Google Form) which can be found here: https://scitools.org.uk/cla/v4/form 

- [ ] @andreas-h
- [x] @ajdawson 
- [x] @mir06 
- [x] @cvelascof 
- [x] @cbcunc 
- [x] @duncanwp 
- [x] @jswanljung 
- [x] @philippjfr 
- [x] @SimonPeatman 
- [ ] @shoyer 
- [x] @bascrezee 

This form is also linked from the SciTools Governance page: https://scitools.org.uk/organisation.html#governance
 
Once all contributors listed above have signed the new CLA we will re-license Iris to BSD-3.

### PR DESCRIPTION
**Pull request by lbdreyer**
Closes #5148

It has now passed the deadline we set in #5148 by when objections to relicensing to BSD needed to be raised.

### CODE REVIEW

**Review by bjlittle — CHANGES_REQUESTED** [review]
@lbdreyer Total stellar effort 🤩

Add a `whatsnew` entry and we're good to roll with this. Hello BSD-3!

Let's just hope we don't relicense anytime soon again 🤣

**Review by bjlittle — APPROVED** [review]

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ This document explains the changes made to Iris for this release
 📢 Announcements
 ================
 
-#. N/A
+#. `lbdreyer`_ relicensed Iris from LGPL-3 to BSD-3. (:pull: `5577`)
```
Comment (bjlittle) [inline docs/src/whatsnew/latest.rst:27]:
```suggestion
#. `@lbdreyer`_ relicensed Iris from LGPL-3 to BSD-3. (:pull: `5577`)
```

Comment (lbdreyer) [reply docs/src/whatsnew/latest.rst]: Thanks for fixing @bjlittle! I don't think I've ever managed to get the formatting of a whatsnew entry correct the first time!!

**Comment by lbdreyer** [comment]
Thanks for reviewing and merging @bjlittle !

**Pull request merged**